securedrop/journalist.py

```diff
@@ -242,9 +242,23 @@ def admin_reset_two_factor_hotp():
     otp_secret = request.form.get('otp_secret', None)
     if otp_secret:
         user = Journalist.query.get(uid)
-        user.set_hotp_secret(otp_secret)
-        db_session.commit()
-        return redirect(url_for('admin_new_user_two_factor', uid=uid))
+        try:
+            user.set_hotp_secret(otp_secret)
+        except TypeError as e:
+            if "Non-hexadecimal digit found" in str(e):
+                flash("Invalid secret format: "
+                      "please only submit letters A-F and numbers 0-9",
+                      "error")
+            elif "Odd-length string" in str(e):
+                flash("Invalid secret format: "
+                      "odd-length secret. Did you mistype the secret?",
+                      "error")
+            else:
+                flash("Unexpected error", "error")
```

Comment:
It would be better to flash an error more like https://github.com/dachary/securedrop/blob/develop/securedrop/journalist.py#L277 here. Also, this still needs logging. In previous iterations where you re-raised the `TypeError` in the `else` block, Apache would log it for us. Now, the exception will disappear, which we'd like to avoid.